packages/runtime-core/src/componentEmits.ts

```diff
@@ -121,7 +121,7 @@ export function emit(
     }Modifiers`
     const { number, trim } = props[modifiersKey] || EMPTY_OBJ
     if (trim) {
-      args = rawArgs.map(a => a.trim())
+      args = rawArgs.map(a => (a.trim ? a.trim() : a))
```

Comment:
I think it would be better to check if `a` is a string. Otherwise this will throw an error:
```
this.$emit('update:modelValue', { trim: true })
```

```suggestion
      args = rawArgs.map(a => (typeof a === 'string' ? a.trim() : a))
```